great_expectations/checkpoint/checkpoint.py

```diff
@@ -911,7 +915,7 @@ def construct_from_config_args(
         config_version: Optional[Union[int, float]] = None,
         template_name: Optional[str] = None,
         module_name: Optional[str] = None,
-        class_name: Optional[str] = None,
+        class_name: str = "Checkpoint",
```

Comment:
Is this a literal or can it be any string?

Reply:
I tried to make it a `Literal` but I started get this warning in `absDataContext`
```
great_expectations/data_context/data_context/abstract_data_context.py:2109: error: Argument "class_name" to "construct_from_config_args" of "Checkpoint" has incompatible type "Optional[str]"; expected "Literal['Checkpoint', 'SimpleCheckpoint']"  [arg-type]
```
Although if that's the only place maybe it's better to just ignore/fix the problem there.

Reply:
Oh actually I'm still getting something very similar so let me take a look. I may add this back.
```
great_expectations/data_context/data_context/abstract_data_context.py:2109: error:
Argument "class_name" to "construct_from_config_args" of "Checkpoint" has
incompatible type "Optional[str]"; expected "str"  [arg-type]
                    class_name=class_name,
                               ^~~~~~~~~~
```

Reply:
I added it back and added an type ignore with a comment in `AbstractDataContext._resolve_add_checkpoint_args()`